app/controllers/collections_controller.rb

```diff
@@ -39,6 +42,61 @@ def show
     end
   end
 
+  def create
+    if request.format == 'json' and request.post?
+      collection_name = params[:name]
+      if (!collection_name.nil? and !collection_name.blank? and !(collection_name.length > 255) and !(params[:collection_metadata].nil?))
+        # Parse JSON-LD formatted collection metadata and convert to RDF
+        json_md = params[:collection_metadata]
+        graph = RDF::Graph.new << JSON::LD::API.toRdf(json_md)
+        collection_rdf = graph.dump(:ttl, prefixes: {foaf: "http://xmlns.com/foaf/0.1/"})
+        collection_uri = graph.statements.first.subject.to_s
+        if !Collection.find_by_uri(collection_uri).present?  # ingest skips collections with non-unique name or uri
+          # write metadata and manifest file for ingest
+          collection_dir = NEW_COLLECTION_DIR
+          FileUtils.mkdir_p(collection_dir) unless File.directory?(collection_dir)
+          metadata_file_path = File.join(collection_dir,  collection_name + '.n3')
+          corpus_dir = File.join(collection_dir, collection_name)
+          FileUtils.mkdir_p(corpus_dir) unless File.directory?(corpus_dir)
+          manifest_file_path = File.join(corpus_dir, MANIFEST_FILE_NAME)
+          begin
+            metadata_file = File.open(metadata_file_path, 'w')
+            metadata_file.puts collection_rdf
+          ensure
+            metadata_file.close if !metadata_file.nil?
+          end
+          begin
+            manifest_hash = {"collection_name" => collection_name, "files" => {}}
+            manifest_file = File.open(manifest_file_path, 'w')
+            manifest_file.puts(manifest_hash.to_json)
+          ensure
+            manifest_file.close if !manifest_file.nil?
+          end
```

Comment:
There's a much cleaner way of dealing with files in Ruby, just use `File.open|` with a block, it does pretty much what you are doing with those `begin` and `ensure` blocks and is part of the standard library. Examlpe taken from this [SO](https://stackoverflow.com/questions/2191632/begin-rescue-and-ensure-in-ruby):

```
File.open('myFile.txt', 'w') do |file|
  file.puts content
end
```

Which is implemented something like:

```
def File.open(filename, mode='r', perm=nil, opt=nil)
  yield filehandle = new(path, mode, perm, opt)
ensure
  filehandle.close
end
```